config/initializers/filter_parameter_logging.rb

```diff
@@ -1,4 +1,32 @@
 # Be sure to restart your server when you modify this file.
+module ParameterFiltering
+  ATTRIBUTE_WHITELIST = %w[
+    action
+    active
+    code
+    controller
+    created_at
+    deleted_at
+    limit
+    metadata_version
+    process_token
+    recorded_at
+    status
+    updated_at
+  ].freeze
 
-# Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+  WHITELIST_REGEX = /(^|_)ids?|#{Regexp.union(ATTRIBUTE_WHITELIST)}/.freeze
+  SANITIZED_VALUE = "[FILTERED]".freeze
+end
+
+Rails.application.config.filter_parameters += [:password, :age]
+Rails.application.config.filter_parameters << lambda do |key, value|
+  unless key.match(ParameterFiltering::WHITELIST_REGEX)
+    case value
+    when String
+      value.replace(ParameterFiltering::SANITIZED_VALUE)
+    else
+      value.to_s.replace(ParameterFiltering::SANITIZED_VALUE)
+    end
+  end
```

Comment:
Hmm, rather than using the module as a bucket of constants, could we have the module generate this lambda for us as well?

```ruby
module ParameterFiletering
  # the constants...

  def self.filter
    lambda do |key, value|
      # the magic...
    end
  end
end

Rails.application.config.filter_parameters += [:password, :age, ParameterFiltering.filter]
```

We could also unit test this module effectively.

Reply:
Done. Yeah its nicer having it all in the module.

I'm not too concerned with unit testing this for now, as the request spec covers that this works end-to-end w/ the logging system.